---
title: "jul19_blast_filter"
author: "Hannah Reich"
date: "July 22, 2019"
output: html_document
---

```{r}
# jul 22 2019
# this script is to filter blast output to triple check Sfitti assembly for coral contamination

# blasted against smic, stridac, 3 acropora
# smic abbrev for V2 = Smic.scaffold
# stridac abbrev for V2 = scaffold
# Ahyan abbrev for V2 = flattened
# Aten abbrev for V2 = Sc
# Adig abbrev for V2 = NW_
# Acerv abbrev for V2 = Segkk
# Apalm abbrev for V2 = Sc0a5M3

# set the working directory
setwd("~/Desktop/PhD/Symbiodinium fitti/blast reports/")

# load packages
library(dplyr)
library(tidyverse)

# read in the data
# coral contam was wrangled into 1 text file on the aci server using awk. script = jul19_extra_blast.sh
max1 <- read.table("~/Desktop/PhD/Symbiodinium fitti/blast reports/coral_contam_max1.txt")
max3 <- read.table("~/Desktop/PhD/Symbiodinium fitti/blast reports/coral_contam_max3.txt")

# make the column names useful
# somewhat matches blast output: qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle
colnames(max1) <- c("site", "coral_scaf", "pident", "ident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore")
colnames(max3) <- c("site", "coral_scaf", "pident", "ident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore")

# remove scaffold duplicates
# useful site: https://www.datanovia.com/en/lessons/identify-and-remove-duplicate-data-in-r/
distinct_max1 <-max1 %>% distinct(site, .keep_all = TRUE)
count(distinct_max1) # went from 114 to 39 scaffolds (kept 34%)

distinct_max3 <-max3 %>% distinct(site, .keep_all = TRUE)
count(distinct_max3) # went from 5918 to 1598 scaffolds (kept 27%)

# use the intersect function to find rows that appear in both datasets and those that are unique to the max3 dataset
# this was useful: https://www.rstudio.com/wp-content/uploads/2015/02/data-wrangling-cheatsheet.pdf
all <- intersect(distinct_max3, distinct_max1)
count(all) # 39 scaffolds

dif <- setdiff(distinct_max3, distinct_max1)
count(dif) # 1559 scaffolds

# time to see what genes and how many SNPs/indels are on these scaffolds.
# read in the genome annotation file
annot <- read.csv("~/Desktop/PhD/Symbiodinium fitti/blast reports/combinedblast04112018.csv", header = T)
genebed <- read.table("~/Desktop/PhD/Symbiodinium fitti/blast reports/sfitti_genes.bed", header = T) # this has the scaffold ID stuff
info <- merge(annot, genebed, by = "Protein_ID") # links scaffold ID to genome annotation, needed
snps <- read.table("~/Desktop/PhD/Symbiodinium fitti/SNP_under_selection/v_halloween2018/halloween_mm80.tab", header = T)

# use a right_join() to pull out the genes on the scaffolds with contam
contam <- right_join(distinct_max3, info, by = "site")
# extract genes that don't hit to coral scaf
real_contam <- contam %>% drop_na(coral_scaf)
count(real_contam) # this affects 342 genes

# figure out how many snps this will effect
snp_contam <- merge(snps, distinct_max3, by.x = "CHROM", by.y = "site")
count(snp_contam) # this affects 7087 SNPS

# write out txt files to share
write_csv(real_contam, "genes_contam_jul19.csv") # genes of concern
write_csv(distinct_max3, "scaffolds_contam_jul19.csv") # scaffolds of concern

```

