---
title: "jul19_blast_filter"
author: "Hannah Reich"
date: "July 22, 2019"
output: html_document
---

```{r}
# jul 22 2019
# this script is to filter blast output to triple check Sfitti assembly for coral contamination

# blasted against smic, stridac, 3 acropora
# smic abbrev for V2 = Smic.scaffold
# stridac abbrev for V2 = scaffold
# Ahyan abbrev for V2 = flattened
# Aten abbrev for V2 = Sc
# Adig abbrev for V2 = NW_
# Acerv abbrev for V2 = Segkk
# Apalm abbrev for V2 = Sc0a5M3

# set the working directory
setwd("~/Desktop/PhD/Symbiodinium fitti/blast reports/")

# load packages
library(dplyr)
library(tidyverse)

# read in the data
# coral contam was wrangled into 1 text file on the aci server using awk. script = jul19_extra_blast.sh
max1 <- read.table("~/Desktop/PhD/Symbiodinium fitti/blast reports/coral_contam_max1.txt")
max3 <- read.table("~/Desktop/PhD/Symbiodinium fitti/blast reports/coral_contam_max3.txt")

# make the column names useful
# somewhat matches blast output: qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle
colnames(max1) <- c("site", "coral_scaf", "pident", "ident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore")
colnames(max3) <- c("site", "coral_scaf", "pident", "ident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore")

# remove scaffold duplicates
# useful site: https://www.datanovia.com/en/lessons/identify-and-remove-duplicate-data-in-r/
distinct_max1 <-max1 %>% distinct(site, .keep_all = TRUE)
count(distinct_max1) # went from 114 to 39 scaffolds (kept 34%)

distinct_max3 <-max3 %>% distinct(site, .keep_all = TRUE)
count(distinct_max3) # went from 5918 to 1598 scaffolds (kept 27%)

# use the intersect function to find rows that appear in both datasets and those that are unique to the max3 dataset
# this was useful: https://www.rstudio.com/wp-content/uploads/2015/02/data-wrangling-cheatsheet.pdf
all <- intersect(distinct_max3, distinct_max1)
count(all) # 39 scaffolds

dif <- setdiff(distinct_max3, distinct_max1)
count(dif) # 1559 scaffolds

# time to see what genes and how many SNPs/indels are on these scaffolds.
# read in the genome annotation file
annot <- read.csv("~/Desktop/PhD/Symbiodinium fitti/blast reports/combinedblast04112018.csv", header = T)
genebed <- read.table("~/Desktop/PhD/Symbiodinium fitti/blast reports/sfitti_genes.bed", header = T) # this has the scaffold ID stuff
info <- merge(annot, genebed, by = "Protein_ID") # links scaffold ID to genome annotation, needed
snps <- read.table("~/Desktop/PhD/Symbiodinium fitti/SNP_under_selection/v_halloween2018/halloween_mm80.tab", header = T)

#### max-target = 3
# use a right_join() to pull out the genes on the scaffolds with contam
contam <- right_join(distinct_max3, info, by = "site")
# extract genes that don't hit to coral scaf
real_contam <- contam %>% drop_na(coral_scaf)
count(real_contam) # this affects 342 genes

# figure out how many snps this will effect
snp_contam <- merge(snps, distinct_max3, by.x = "CHROM", by.y = "site")
count(snp_contam) # this affects 7087 SNPS

#### max-target = 1
# use a right_join() to pull out the genes on the scaffolds with contam
contam <- right_join(distinct_max1, info, by = "site")
# extract genes that don't hit to coral scaf
real_contam <- contam %>% drop_na(coral_scaf)
count(real_contam) # this affects 7 genes

# figure out how many snps this will effect
snp_contam <- merge(snps, distinct_max1, by.x = "CHROM", by.y = "site")
count(snp_contam) # this affects 486 SNPS

# write out txt files to share
#write_csv(real_contam, "genes_contam_jul19.csv") # genes of concern
#write_csv(distinct_max3, "scaffolds_contam_jul19.csv") # scaffolds of concern


# aug 2 2019 update
# add in scaffolds from sheila
# also add in coral blast results

# add in coral only blast search
cmax1 <- read.table("~/Desktop/PhD/Symbiodinium fitti/blast reports/coral_max1.txt")
cmax3 <- read.table("~/Desktop/PhD/Symbiodinium fitti/blast reports/coral_max3.txt")

# make the column names useful
# somewhat matches blast output: qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle
colnames(cmax1) <- c("site", "coral_scaf", "pident", "ident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore")
colnames(cmax3) <- c("site", "coral_scaf", "pident", "ident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore")

# remove scaffold duplicates
distinct_cmax1 <-cmax1 %>% distinct(site, .keep_all = TRUE)
count(distinct_cmax1) # went from 55,766 to 9,313 scaffolds (kept 16.7%)

distinct_cmax3 <-cmax3 %>% distinct(site, .keep_all = TRUE)
count(distinct_cmax3) # went from 134,300 to 9,313 scaffolds (kept 6.93%)
# woohoo, same # of scaffolds from both blast searches

# use the intersect function to find rows that appear in both datasets and those that are unique to the cmax3 dataset
allc <- intersect(distinct_cmax3, distinct_cmax1)
count(allc) # 9,313 scaffolds

difc <- setdiff(distinct_cmax3, distinct_cmax1)
count(difc) # 0 scaffolds
# 0 different scaffolds between the two, only need to use the max target seq = 1 blast search.

# find the intersection from the both blast searches
allboth <- intersect(distinct_max3, distinct_cmax1)
count(allboth) # 238 shared scaffolds

# compare coral blast search against holobiont blast search
dif_coral <- setdiff(distinct_cmax1, distinct_max3)
count(dif_coral) # 9075 different scaffolds
# (9313 coral blast scaffolds = 9075 unique scaffolds from coral blast + 238 scaffolds shared with holobiont blast)

# compare holobiont blast search against coral blast search
dif_holo <- setdiff(distinct_max3, distinct_cmax1)
count(dif_holo) # 1360 different scaffolds 
# (1598 holobiont blast scaffolds = 1360 unique scaffolds from holobiont blast + 238 scaffolds shared with coral blast)

# create columns describing the blast searches for each dataset
dif_coral$search <- c("Coral_only_blast")
dif_holo$search <- c("Holobiont_only_blast")
allboth$search <- c("Coral_AND_holobiont_blast")

# join the three search categories
all_searches <- rbind(dif_holo, dif_coral, allboth)
distinct_all <- all_searches %>% distinct(site, .keep_all = TRUE)
count(distinct_all) # 9313 scaffolds

# figure out how many genes and snps this affects
# use a right_join() to pull out the genes on the scaffolds with contam
contam_both <- right_join(distinct_all, info, by = "site")
# extract genes that don't hit to coral scaf
real_contam_both <- contam_both %>% drop_na(coral_scaf)
count(real_contam_both) # this affects 956 genes

# figure out how many snps this will effect
snp_contam_both <- merge(snps, distinct_all, by.x = "CHROM", by.y = "site")
count(snp_contam_both) # this affects 10,246 SNPS

############
############
# now pair in Sheila's suspicious scaffolds
# removed extra characters the super lazy way in excel
sak <- read.table("~/Desktop/PhD/Symbiodinium fitti/blast reports/SAKcontam_05282019.txt")
colnames(sak) <- c("site")
count(sak) # 210 scaffolds

# use an full-join to add in SAK scaffolds that didn't pop up in the blast searches
sak_and_blast <- full_join(sak, all_searches)

# find the overlap
as_siteonly <- all_searches[,1]
as_siteonly <- as.data.frame(as_siteonly)
colnames(as_siteonly) <- c("site")
sak_blast <- intersect(as_siteonly, sak) # 115 of 210 scaffolds are shared
# there are only 115 distinct scaffolds and they are all shared
distinct_sak <- sak %>% distinct(site, .keep_all = TRUE)

# make sure the sak_and_blast table doesn't have duplicates
distinct_sb <- sak_and_blast %>% distinct(site, .keep_all = TRUE)
# back down to 9313 scafolds

# figure out how many genes and snps this affects
# use a right_join() to pull out the genes on the scaffolds with contam
contam_sb<- right_join(distinct_sb, info, by = "site")
# extract genes that don't hit to coral scaf
real_contam_sb <- contam_sb %>% drop_na(coral_scaf)
count(real_contam_sb) # this affects 956 genes

# figure out how many snps this will effect
snp_contam_sb <- merge(snps, distinct_sb, by.x = "CHROM", by.y = "site")
count(snp_contam_sb) # this affects 10,426 SNPS

####
# compare the SAK subset to the holobiont blast results
# use an full-join to add in SAK scaffolds that didn't pop up in the HOLOBIONT blast searches

##
# max target = 1
sak_hb1 <- full_join(sak, distinct_max1)
d_sak_hb1 <- sak_hb1 %>% distinct(site, .keep_all = TRUE)
count(d_sak_hb1) # 151 scaffolds

# figure out how many genes and snps this affects
# use a right_join() to pull out the genes on the scaffolds with contam
contam_shb1 <- right_join(d_sak_hb1, info, by = "site")
# extract genes that don't hit to coral scaf
real_contam_shb1 <- contam_shb1 %>% drop_na(coral_scaf)
count(real_contam_shb1) # this affects 7 genes

# figure out how many snps this will effect
snp_contam_shb1 <- merge(snps, d_sak_hb1, by.x = "CHROM", by.y = "site")
count(snp_contam_shb1) # this affects 2409 SNPS

##
# max target = 3
sak_hb3 <- full_join(sak, distinct_max3)
d_sak_hb3 <- sak_hb3 %>% distinct(site, .keep_all = TRUE)
count(d_sak_hb3) # 1641 scaffolds

# figure out how many genes and snps this affects
# use a right_join() to pull out the genes on the scaffolds with contam
contam_shb3 <- right_join(d_sak_hb3, info, by = "site")
# extract genes that don't hit to coral scaf
real_contam_shb3 <- contam_shb3 %>% drop_na(coral_scaf)
count(real_contam_shb3) # this affects 342 genes

# figure out how many snps this will effect
snp_contam_shb3 <- merge(snps, d_sak_hb3, by.x = "CHROM", by.y = "site")
count(snp_contam_shb3) # this affects 7710 SNPS


### aug 6 2019
# figure out how many scaffolds don't hit to sym or coral
# see how many genes & snps this effects too

# load in the data of blast hits
all <- read.table("~/Desktop/PhD/Symbiodinium fitti/blast reports/symORcoral_max1.txt")

# rename columns 
colnames(all) <- c("site", "scaf", "pident", "ident", "length", "mismatch", "gapopen", "qstart", "qend", "sstart", "send", "evalue", "bitscore")
count(all) # 1,048,576 scaffolds

# remove duplicates
distinct_all <-all %>% distinct(site, .keep_all = TRUE)
count(distinct_all) # 195,129 scaffolds hit to sym or coral
distinct_all_siteonly <- distinct_all[,1]
distinct_all_siteonly <- as.data.frame(distinct_all_siteonly)
colnames(distinct_all_siteonly) <- c("site")

# load in the file of all scaffold names
scafs <- read.table("~/Desktop/PhD/Symbiodinium fitti/blast reports/scaf_names_sfitti.txt")

colnames(scafs) <- c("site")

# find the scaffolds that did not hit to sym or coral
rebels <- setdiff(scafs, distinct_all_siteonly)
count(rebels) # 79,056

# figure out how many genes and snps this affects
# use a right_join() to pull out the genes on the scaffolds with contam
rebel_genes <- right_join(info, rebels, by = "site")
# extract genes that don't hit to coral scaf
real_contam_genes <- rebel_genes %>% drop_na(Protein_ID)
count(real_contam_genes) # this affects 2,667 genes

# figure out how many snps this will effect
snp_contam_rebels <- merge(snps, rebels, by.x = "CHROM", by.y = "site")
count(snp_contam_rebels) # this affects 10,574 SNPS

```

```{r eval=FALSE, include=FALSE}
# counting number of scaffolds on server

# count the opera scaffolds, there are 11 lines
grep -n "opera_scaffold_*" AC_Sfitti_2.fa > opera_scaf_id.txt

# count the number of regular scaffolds
# there are 274,174 lines
grep -n "size" AC_Sfitti_2.fa > reg_scaf_id.txt

# grep out all of the scaffold names for comparison
grep -e "scaffold" AC_Sfitti_2.fa > scaf_names_sfitti_extrachar.txt

# remove the ">" character
cat scaf_names_sfitti_extrachar.txt | tr -d '>' > scaf_names_sfitti.txt
```

